### ISSUE TITLE
Issues handling resize event

### ISSUE THREAD

**Issue by richardellwood** [termbox/termbox2#35]
I am writing an interactive program that constantly calls tb_peek_event(&event, 0) over and over in a loop.  When I get a event.type == TB_EVENT_RESIZE, I capture event.w and event.h and call a function to redraw everything.

(Note, if I get a DB_ERR_POLL I check to see if db_last_errno() is EINTR and if so, call tb_peek_event() again as explained in the docs.)

However, when growing the terminal window either horizontally or vertically, when I try to set a cell in an area of the window that has newly appeared due to the larger size, I get an Out of Bounds error.

Furthermore, I've noticed that the event.w and event.h that I captured above at the time of the event is the old size of the frame not the new size.

**Comment by adsr** [comment]
Hi @richardellwood, can you make a small repro case? This example seems to work:

```c
#define TB_IMPL
#include "termbox.h"

int main(int argc, char **argv) {
    struct tb_event e;
    int rv, w, h;

    tb_init();
    w = tb_width();
    h = tb_height();

    rv = 0;
    while (1) {
        rv = tb_peek_event(&e, 0);

        if (rv == TB_OK) {
            if (e.type == TB_EVENT_KEY && e.ch == 'q') {
                break;
            }

            if (e.type == TB_EVENT_RESIZE) {
                w = e.w;
                h = e.h;
            }
        } else if (rv != TB_ERR_NO_EVENT && rv != TB_ERR_POLL) {
            break;
        }

        int x, y;
        for (y = 0; y < h; y++) {
            for (x = 0; x < w; x++) {
                tb_print(x, y, 0, 0, "a");
            }
        }
        tb_present();
    }

    tb_shutdown();

    if (rv < 0) {
        rv *= -1;
    }
    return rv;
}
```

**Comment by richardellwood** [comment]
From my tests, that sample produces the error if you modify the code to test for it.  I modified the following section.

```
    int x, y, err;
    for (y = 0; y < h; y++) {
      for (x = 0; x < w; x++) {
        err = tb_print(x, y, 0, 0, "a");
        if(err != TB_OK) {
          tb_shutdown();
          printf("%d %s\n", err, tb_strerror(err));
          return err;
        }
      }
    }
```
Then when I resize the terminal window to make it larger the program ends with `-9 Out of bounds`.

**Comment by adsr** [comment]
Thanks for the test programs. I believe this fixes: https://github.com/termbox/termbox2/pull/38.

**Comment by adsr** [comment]
Fixed in 7c84bca93d5816383854c663148bab94de27a0bb

**Comment by apprehensions** [comment]
Currently having this issue with a simple program and the keyboard demo program.

**Comment by adsr** [comment]
Hi @apprehensions, can you share your simple program?

On my setup, the demo program was exiting on resize due to an unhandled `EINTR`. I fixed this in 4d1c41e.

**Comment by apprehensions** [comment]
Yep, that solves it, thanks!
